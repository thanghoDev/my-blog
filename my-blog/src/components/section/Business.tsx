import React from 'react';
import Image from 'next/future/image';
import { Col } from 'react-bootstrap';
import { Blog } from '@/types/blog';

type BusinessProps = {
  business: Blog[];
};

function Business({ business }: BusinessProps) {
  return (
    <Col md={6}>
      <h2 className='mb-5 text-decoration-underline fs-5'>Business</h2>
      {business.map((item) => (
        <div key={item.id} className='d-flex mb-3'>
          <Image width='165' height='180' src={item.images} alt={item.alt} />
          <div className='ms-4'>
            <h2 className='fs-6 text-black text-capitalize'>{item.title}</h2>
            <p className='description'>{item.description}</p>
            <p className='text-capitalize category'>
              Dave Rogers <code className='text-secondary'>in</code>{' '}
              {item.category}
            </p>
          </div>
        </div>
      ))}
    </Col>
  );
}

export default Business;
