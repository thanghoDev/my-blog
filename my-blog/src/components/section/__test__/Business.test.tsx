import React from 'react';
import Business from '../Business';

import { render, screen } from '@testing-library/react';

describe('testing business section', () => {
  const business = [
    {
      id: 1,
      title: 'title',
      description: 'description',
      category: 'business',
      view: 150,
      images: 'https://i.ibb.co/10hj3h1/business.jpg',
      alt: 'business',
    },
  ];
  test('render section', () => {
    const { container } = render(<Business business={business} />);
    expect(container.firstChild).toMatchSnapshot();
    const heading = screen.getByRole('heading', {
      name: /business/i,
    });
    expect(heading).toBeInTheDocument();

    const image = screen.getByRole('img', {
      name: 'business',
    });
    expect(image).toBeInTheDocument();

    const title = screen.getByRole('heading', {
      name: 'title',
    });
    expect(title).toBeInTheDocument();
  });
});
