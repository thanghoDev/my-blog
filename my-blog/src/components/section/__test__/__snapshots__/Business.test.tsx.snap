// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing business section render section 1`] = `
<div
  class="col-md-6"
>
  <h2
    class="mb-5 text-decoration-underline fs-5"
  >
    Business
  </h2>
  <div
    class="d-flex mb-3"
  >
    <img
      alt="business"
      data-nimg="future"
      decoding="async"
      height="180"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fbusiness.jpg&w=384&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fbusiness.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2Fbusiness.jpg&w=384&q=75 2x"
      style="color: transparent;"
      width="165"
    />
    <div
      class="ms-4"
    >
      <h2
        class="fs-6 text-black text-capitalize"
      >
        title
      </h2>
      <p
        class="description"
      >
        description
      </p>
      <p
        class="text-capitalize category"
      >
        Dave Rogers 
        <code
          class="text-secondary"
        >
          in
        </code>
         
        business
      </p>
    </div>
  </div>
</div>
`;
